src/ring_theory/integral_domain.lean

```diff
@@ -131,6 +131,29 @@ end
 
 end
 
+section euclidean_division
+
+namespace polynomial
+
+open_locale polynomial
+
+variables (K : Type) [field K] [algebra R[X] K]  [is_fraction_ring R[X] K]
+
+lemma div_eq_quo_add_rem_div (f : R[X]) {g : R[X]} (hg : g.monic) :
+  ∃ q r : R[X], r.degree < g.degree ∧ (↑f : K) / ↑g = ↑q + ↑r / ↑g :=
```

Comment:
Surprised that we don't have `f = f /ₘ g * g + f %ₘ g`

Reply:
I've added in a `lemma exists_quotient_remainder` and proven it as well. I didn't use it to prove the fraction ring case because I'm not sure if that's necessary... what do you recommend?
Also @alreadydone that seems similar to `mod_by_monic_add_div`

Reply:
Oh yes, it literally appears in your proof and I don't know why I ignored it.
Given that, I think `exists_quotient_remainder` is not that necessary since it just combines two lemmas`degree_mod_by_monic_lt` and `mod_by_monic_add_div` and change multiplication order in the latter. 
This proof is mostly about coercion it seems.

Reply:
Yes, I see what you mean... in fact, wouldn't `mod_by_monic_add_div` be stronger, since it uses `f /ₘ g` and `f %ₘ g` specifically while `exists_quotient_remainder` just says a quotient and remainder exist without specifying what they are? Still, maybe it could be useful to combine `degree_mod_by_monic_lt` and `mod_by_monic_add_div` into a single result. I'm not sure...

Reply:
I like this theorem because the statement essentially says that `polynomial R` is "nearly" a Euclidean domain, except the divisors must be monic. So it feels nice to have a single theorem stating that, rather than having to put it together from `degree_mod_by_monic_lt` + `mod_by_monic_add_div`.

Reply:
@Vierkantor I think you replied somewhere else that this lemma tells you that R[X] is nearly a Euclidean Domain (wasn't able to find that thread) and I think that while that does make sense, it's not necessary to include it in mathlib because it's a combination of two existing results... besides, stating it as an abstract existence proof might be a bit weaker than providing specific values for `q` and `r` (which is what we have in `mod_by_monic_add_div`). What do you think?

Reply:
(I see [the message you're referring to](https://github.com/leanprover-community/mathlib/pull/17709/files#r1063494393) just above your reply in the same thread.)

I agree that the proof of the result is rather elementary and by itself would not warrant inclusion. Still I feel there's quite a bit of worth in providing these milestones/guideposts for those searching the library. It's not such an important issue to me though, so if you hate `exists_quotient_remainder` then please revert it :)